API/Controllers/UsersController.cs

```diff
@@ -0,0 +1,149 @@
+    /// </summary>
+    /// <param name="id">Id of the user</param>
+    /// <response code="404">If a user does not exist</response>
+    /// <response code="200">Returns a list of submissions</response>
+    [HttpGet("{id:int}/submissions")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType<IEnumerable<MatchSubmissionStatusDTO>>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> GetSubmissionsAsync(int id)
+    {
+        if (!await _userService.ExistsAsync(id))
+        {
+            return NotFound();
+        }
+
+        return Ok(await _userService.GetSubmissionsAsync(id) ?? new List<MatchSubmissionStatusDTO>());
+    }
+
+    /// <summary>
+    /// Rejects a user's match submissions
+    /// </summary>
+    /// <param name="id">Id of the user</param>
+    /// <response code="404">If a user does not exist</response>
+    /// <response code="200">Denotes the operation was successful</response>
+    [HttpPost("{id:int}/submissions:reject")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> RejectSubmissionsAsync(int id)
+    {
+        if (!await _userService.ExistsAsync(id))
+        {
+            return NotFound();
+        }
+
+        var verifierId = HttpContext.AuthorizedUserIdentity();
+        return await _userService.RejectSubmissionsAsync(id, verifierId, (int)MatchVerificationSource.Admin)
+            ? BadRequest()
+            : Ok();
+    }
+
+    /// <summary>
+    /// Get a user's OAuth clients
+    /// </summary>
+    /// <param name="id">Id of the user</param>
+    /// <response code="404">If a user does not exist</response>
+    /// <response code="200">Returns a list of OAuth clients</response>
+    [HttpGet("{id:int}/clients")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType<IEnumerable<OAuthClientDTO>>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> GetClientsAsync(int id)
+    {
+        if (!await _userService.ExistsAsync(id))
+        {
+            return NotFound();
+        }
+
+        return Ok(await _userService.GetClientsAsync(id) ?? new List<OAuthClientDTO>());
+    }
+
+    /// <summary>
+    /// Delete a user's OAuth client
+    /// </summary>
+    /// <param name="id">Id of the user</param>
+    /// <param name="clientId">Id of the OAuth client</param>
+    /// <response code="404">If a user or client does not exist</response>
+    /// <response code="400">If the deletion was not successful</response>
+    /// <response code="200">Denotes the deletion was successful</response>
+    [HttpDelete("{id:int}/clients/{clientId:int}")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> DeleteClientAsync(int id, int clientId)
```

Comment:
Users should be able to do execute this endpoint. I think the class-level authorization needs to be removed and have each endpoint specify the authorization requirements.